File: src/simplematrix/test_simplematrix.py
#! /usr/bin/env python

from simplematrix import SimpleMatrix

m = SimpleMatrix(100, 100)

i = 0
j = 0
for x in range(16):
    m.entry(i, j, x)
    i += 13; i %= 100
    j += 29; j %= 100

r = m.rank()
assert r == 15
print (r)

File: src/simplematrix/test_simplematrix.cpp
#include <iostream>

#include "simple.hpp"

main(int argc, char **argv)
{
  SimpleMatrix M(100, 100);

  int i = 0;
  int j = 0;
  for (int x = 1; x < 10; x++)
    {
      M.entry(i, j, x);
      i += 13; i %= 100;
      j += 29; j %= 100;
    }

  std::cout << M.rank() << std::endl;
}
